### PR TITLE
Fix waveform "Insert Line", add extra option, and use MMB to scroll waveform

### PR DESCRIPTION
**Pull request by qsniyg**
This commit does the following:

 * When selecting "Insert Line" in the waveform, it'll set it at the wrong index. This fixes it
 * Middle mouse button is used to scroll the waveform, like double clicking does, except it will keep scrolling when MMB is held down. I found this to be much easier.
 * Add "New Line Padding" option, which sets the amount of padding between subtitles when selecting "Insert Before" or "Insert After". The default (100ms) is kept the same, but I added it since I personally prefer a padding of 0.

I added in these features quickly, and so I didn't think to separate them into different commits or branches, let me know if you'd rather I do this.

### CODE REVIEW

File: src/waveformwidget.cpp [hunk 1]
```diff
@@ -1075,10 +1091,11 @@ WaveformWidget::showContextMenu(QMouseEvent *event)
 
 				int insertIndex = 0;
 				foreach(SubtitleLine *sub, m_subtitle->allLines()) {
+					if(sub->showTime() <= timeShow)
```
Comment (maxrd2) [inline src/waveformwidget.cpp]: remove this if please

**Comment by maxrd2** [comment]
Thanks for the contribution. Would prefer if you can just split changes into separate commits.

Could you rename "New Line Padding" to "Default space between subtitle lines" or something that is more obvious, am not sure if new line padding will be obvious.
Also would be great while you're at it to make that 1000 duration configurable too. "Default subtitle line duration" or something. If it's a problem will do it myself.

**Comment by qsniyg** [comment]
I renamed it as "Default duration between subtitle lines" to be consistent with "Default subtitle line duration", but I can rename it to "Default space between subtitle lines" if you prefer that :)

I renamed the NewLinePadding variable to LineMargin, to be consistent with LineDuration (for the second option). I also fixed the issue where adding a new line results in a subtitle of length LineDuration - LineMargin (instead of LineDuration).

**Comment by qsniyg** [comment]
I'm still new to pull requesting, so I'm not quite sure how to do a proper branch, since it does incorporate changes from this PR as well.

With that being said, the commit above fixes #93

**Comment by maxrd2** [comment]
thank you

**Pull request merged**